vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinServletContextInitializer.java

```diff
@@ -355,28 +355,12 @@ public void failFastContextInitialized(ServletContextEvent event) {
             ApplicationRouteRegistry registry = ApplicationRouteRegistry
                     .getInstance(new VaadinServletContext(
                             event.getServletContext()));
-
-            Set<Class<? extends Component>> errorComponents = findBySuperType(
+            Stream<Class<? extends Component>> hasErrorComponents = findBySuperType(
                     getErrorParameterPackages(), HasErrorParameter.class)
                             .filter(Component.class::isAssignableFrom)
-                            // Replace Flow default with custom version for
-                            // Spring
-                            .filter(clazz -> clazz != RouteNotFoundError.class)
-                            .map(clazz -> (Class<? extends Component>) clazz)
-                            .collect(Collectors.toSet());
-
-            // If there is no custom HasErrorParameter<? super
-            // NotFoundException>
-            // add SpringRouteNotFoundError, with Spring Boot specific hints
-            if (errorComponents.stream().noneMatch(clazz -> {
-                Class<?> exceptionType = ReflectTools.getGenericInterfaceType(
-                        clazz, HasErrorParameter.class);
-                return exceptionType != null && exceptionType
-                        .isAssignableFrom(NotFoundException.class);
-            })) {
-                errorComponents.add(SpringRouteNotFoundError.class);
-            }
-            registry.setErrorNavigationTargets(errorComponents);
+                            .map(clazz -> (Class<? extends Component>) clazz);
+            registry.setErrorNavigationTargets(
+                    hasErrorComponents.collect(Collectors.toSet()));
```

Comment:
What code is actually deciding that `SpringRouteNotFoundError` will be used instead of `RouteNotFoundError` ? Since both have `@DefaultErrorHandler`.

Reply:
The choosing code is the default selection as SpringRouteNotFoundError is picked as it extends the RouteNotFoundError so the DefaultHandler annotation is not checked in that case.